pkg/rancher-desktop/backend/containerClient/registry.ts

```diff
@@ -0,0 +1,97 @@
+import registryAuth from '@pkg/backend/containerClient/auth';
+import { parseImageReference } from '@pkg/utils/dockerUtils';
+import fetch, { Headers } from '@pkg/utils/fetch';
+
+/**
+ * Registry interaction, with both Docker Hub and Docker Registry V2 APIs.
+ */
+class DockerRegistry {
+  /**
+   * Fetch some API endpoint from the registry
+   * @param endpoint The API endpoint, including the registry host.
+   */
+  async get(endpoint: URL): ReturnType<typeof fetch> {
+    const headers = await this.authenticate(endpoint);
+
+    return await fetch(endpoint.toString(), { headers });
+  }
+
+  /**
+   * List all tags for the given image name.
+   * @param name An image name, including registry as needed.
+   */
+  async getTags(name: string): Promise<string[]> {
+    const info = parseImageReference(name);
+    const tags: string[] = [];
+
+    if (!info) {
+      throw new Error(`Invalid image name: "${ name }"`);
+    }
+
+    let endpoint = new URL(`/v2/${ info.name }/tags/list?n=65536`, info.registry);
+    let hasMore = true;
+
+    while (hasMore) {
+      const resp = await this.get(endpoint);
+
+      if (!resp.ok) {
+        throw new Error(`Failed to fetch ${ endpoint }: ${ resp.status } ${ resp.statusText }`);
+      }
+
+      const result: { name: string, tags: string[] } = await resp.json();
+
+      if (result.name !== info.name) {
+        throw new Error(`Invalid tags: incorrect response name ${ result.name } from ${ endpoint }`);
+      }
+
+      tags.push(...result.tags);
+      hasMore = false;
+
+      for (const link of resp.headers[getAsList]('Link') ?? []) {
+        const fields = link.split(/;\s*/);
+
+        if (!fields.includes('rel="next"')) {
+          continue;
+        }
+        endpoint = new URL(fields[0].replace(/^<(.+)>$/, '$1'), endpoint);
```

Comment:
This line is key to understanding the `hasMore`-loop. Are there always angle-brackets around the next `rel` value in the headers? Or this in a spec, or is it just for `docker.io` ?

Reply:
This was specified in RFC 8288. I've added a comment about it (and, reading around, made the `rel=next` check case-insensitive and quote-optional, per the RFC).